tap_mambu/sync.py

```diff
@@ -141,12 +141,17 @@ def sync_endpoint(client, #pylint: disable=too-many-branches
     # Get the latest bookmark for the stream and set the last_integer/datetime
     last_datetime = None
     last_integer = None
-    max_bookmark_value = None
+    number_last_occurrence = 0
     if bookmark_type == 'integer':
         last_integer = get_bookmark(state, stream_name, sub_type, 0)
         max_bookmark_value = last_integer
     else:
-        last_datetime = get_bookmark(state, stream_name, sub_type, start_date)
+        if stream_name == 'audit_trail':
+            audit_trail_bookmark = get_bookmark(state, stream_name, sub_type, [start_date, 0])
```

Comment:
I wonder if it might be more extensible to use a dictionary instead of a list here. I imagine a scenario where this stream evolves further and other sorts of state transformations might be required, a dictionary could be easier to reason about in that case and keep that option manageable for future changes.

Maybe something like this?
```
{"last_datetime": "<datetime_value>", "number_last_occurrence": 123}
```

Reply:
Good idea! We considered this case in the first place, but we opted for the simpler implementation for now, because it was a quick fix strictly for the audit stream. We do, however, plan to include this change in one of our future PRs; PR that will have more refactoring, including keeping the bookmark state format extensible for more streams, not just audit trail.